File: NoteApi/src/main/java/lt/viko/eif/mobiliosiosprogrameles/noteapi/model/Category.java
package lt.viko.eif.mobiliosiosprogrameles.noteapi.model;

public class Category {

    private int id;
    private String name;
    private String iconUrl;

    public Category() {}
    public Category(int id, String name, String iconUrl) {
        this.id = id;
        this.name = name;
        this.iconUrl = iconUrl;
    }

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getIconUrl() {
        return iconUrl;
    }
    public void setIconUrl(String iconUrl) {
        this.iconUrl = iconUrl;
    }
}

File: NoteApp/app/src/main/java/lt/viko/eif/mobiliosiosprogrameles/noteapp/domain/model/Category.kt
package lt.viko.eif.mobiliosiosprogrameles.noteapp.domain.model

import lt.viko.eif.mobiliosiosprogrameles.noteapp.data.local.category.CategoryEntity
import lt.viko.eif.mobiliosiosprogrameles.noteapp.data.remote.note.CategoryDto

data class Category(
    val id: Int? = null,
    val name: String? = null,
    val iconUrl: String? = null
) {
    fun toCategoryEntity(): CategoryEntity {
        return CategoryEntity(
            categoryId = id,
            name = name!!,
            iconUrl = iconUrl
        )
    }

    fun toCategoryDto(): CategoryDto {
        return CategoryDto(
            id = id!!,
            name = name!!,
            iconUrl = iconUrl
        )
    }
}

class InvalidCategoryException(message: String) : Exception(message)